semanticdb/semanticdb/specification.md

```diff
@@ -467,72 +451,99 @@ message UniversalType {
-Both type parameters and declarations are modelled by a [Scope](#scope).
+`ByNameType` represents a signature of a by-name parameter.
 
 ```protobuf
-message MethodType {
-  reserved 1, 2;
-  Scope type_parameters = 4;
-  repeated Scope parameterLists = 5;
-  Type return_type = 3;
+message RepeatedType {
+  Type tpe = 1;
 }
 ```
 
-`MethodType` represents a signature of a method, a constructor or the like.
-It features `type_parameters`, `parameterLists` and a `return_type`. Both type
-parameters and parameters are modelled by [Scopes](#scope).
-Moreover, in order to support multiple parameter lists in Scala methods,
-`parameterLists` is a list of lists.
+`RepeatedType` represents a signature of a repeated parameter.
+
+### Signature
 
 ```protobuf
-message ByNameType {
-  Type tpe = 1;
+message Signature {
+  oneof sealed_value {
+    ClassSignature classSignature = 1;
+    MethodSignature methodSignature = 2;
+    TypeSignature typeSignature = 3;
+    ValueSignature valueSignature = 4;
+  }
 }
 ```
 
-`ByNameType` represents a signature of a by-name parameter.
+`Signature` represents definition signatures. Expression types are modelled
+with [Type](#type).
+
+The SemanticDB type system is a superset of the type systems of supported
+languages - currently modelled after the Scala type system [\[18\]][18].
+This section describes the model, while [Languages](#languages) elaborates
+on how language types map onto this model.
 
 ```protobuf
-message RepeatedType {
-  Type tpe = 1;
+message ClassSignature {
+  Scope type_parameters = 1;
+  repeated Type parents = 2;
+  Scope declarations = 3;
 }
 ```
 
-`RepeatedType` represents a signature of a repeated parameter.
+`ClassSignature` represents a signature of a class, a trait or the like.
+Both type parameters and declarations are modelled by a [Scope](#scope).
 
 ```protobuf
-message TypeType {
-  reserved 1;
-  Scope type_parameters = 4;
+message MethodSignature {
+  Scope type_parameters = 1;
+  repeated Scope parameterLists = 2;
+  Type return_type = 3;
+}
+```
+
+`MethodSignature` represents a signature of a method, a constructor or the like.
+It features `type_parameters`, `parameterLists` and a `return_type`. Both type
+parameters and parameters are modelled by [Scopes](#scope).
+Moreover, in order to support multiple parameter lists in Scala methods,
+`parameterLists` is a list of lists.
+
+```protobuf
+message TypeSignature {
+  Scope type_parameters = 1;
   Type lower_bound = 2;
   Type upper_bound = 3;
 }
 ```
 
-`TypeType` represents a signature of a type parameter or a type member.
+`TypeSignature` represents a signature of a type parameter or a type member.
 It features `type_parameters` as well as `lower_bound` and `upper_bound`.
 Type parameters are modelled by a [Scope](#scope).
 
+```protobuf
+message ValueSignature {
+  Type tpe = 1;
+}
+```
+
+`ValueSignature` represents a signature of a field, a parameter or the like.
```

Comment:
Does this mean `ValueSignature` is used for type parameters?

Reply:
Nope. Type parameters use `TypeSignature`. I'll update the docs to exhaustively list supported kinds.

Reply:
https://github.com/scalameta/scalameta/pull/1620